### PR TITLE
Test the same cable drivers as consuming projects

### PR DESCRIPTION
**Pull request by dfarrell07**
In E2E CI for consuming projects, which is non-default and only
triggered if a label is explicitly added, add cable driver to the test
matrix. Only test exactly the combinations that are covered in the CI of
the consuming projects.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Review by skitt — APPROVED** [review]

File: .github/workflows/consuming.yml [hunk 1]
```diff
@@ -18,17 +18,32 @@ jobs:
       matrix:
         project: ['admiral', 'submariner', 'submariner-operator', 'lighthouse']
         deploytool: ['operator', 'helm']
+        cabledriver: ['libreswan']
```
Comment (mkolesnik) [inline .github/workflows/consuming.yml:21]: I think this isn't needed, can you try without it?

**Review by mkolesnik — CHANGES_REQUESTED** [review]

File: .github/workflows/consuming.yml [hunk 2]
```diff
@@ -18,17 +18,32 @@ jobs:
...
         exclude:
           # Admiral E2E doesn't respect deploy-tool params, as it uses clusters without Submariner
           - project: admiral
             deploytool: helm
           # Operator and Helm are mutually exclusive, don't try to use Helm in Operator repo
           - project: submariner-operator
             deploytool: helm
+        include:
+          # Parallel the consuming projects' test matrices for cable driver CI coverage
```
Comment (mkolesnik) [inline .github/workflows/consuming.yml:30]: I'm not sure what this means..
From the name of the PR I thought it's about running in parallel?

Perhaps something like:
"Test additional combinations of cable drivers on select consuming projects"

Comment (dfarrell07) [reply .github/workflows/consuming.yml:30]: It means we test the same combinations here as the consuming projects do in their CI. They aren't just additional, they make the two sets of tests equal. I can try to reword it to be more clear.

File: .github/workflows/consuming.yml (hunk 1 above)
Comment (dfarrell07) [reply .github/workflows/consuming.yml:21]: I did try without, it is required the matrix var exists before `include`ing it.

**Review by nyechiel — APPROVED** [review]

File: .github/workflows/consuming.yml (hunk 1 above)
Comment (mkolesnik) [reply .github/workflows/consuming.yml:21]: OK good to know since in their guide they seem to indicate you can create vars simply by including them

**Review by mkolesnik — APPROVED** [review]
